### ISSUE TITLE
Possible Issue with Recent Reauth Changes

### ISSUE THREAD

**Issue by jtopjian** [gophercloud/gophercloud#855]
Since https://github.com/gophercloud/gophercloud/pull/743 was merged, we've been seeing random unit test failures both in Travis and OpenLab:

```
2018-03-29 17:32:32.861496 | ubuntu-xenial-ut | === RUN   TestConcurrentReauth
2018-03-29 17:32:32.861551 | ubuntu-xenial-ut | --- FAIL: TestConcurrentReauth (18.02s)
2018-03-29 17:32:32.861678 | ubuntu-xenial-ut | 	convenience.go:35: [1;31mFailure in provider_client_test.go, line 120: expected [0m[1;32m1[0m[1;31m but got [0m[1;33m18[0m[1;31m[0m
```

(http://logs.openlabtesting.org/logs/49/849/2bbea63b3dff3e4c7d723da04d4694075e7c815f/check/gophercloud-unittest/ebe21e3/job-output.txt.gz)

The test will pass upon retry, so this feels like a true concurrency issue.

**Comment by jrperritt** [comment]
I'll look into this.

**Comment by jrperritt** [comment]
I can't reproduce this. I've run `go test -v -run Reauth* -count 20 github.com/gophercloud/gophercloud/testing/...` several time,s and they've all passed.

**Comment by jtopjian** [comment]
Yeah, and I haven't seen any unit test failures from Travis or OpenLab for a few days now... bizarre. If it happens with Travis again, I'll make sure I don't re-run the test. Not sure what good that'll do, though.

**Comment by jtopjian** [comment]
@jrperritt Here's a recent Travis failure:

https://travis-ci.org/gophercloud/gophercloud/jobs/362271442

**Comment by jrperritt** [comment]
Thanks, that helps. You can re-run that one if you want. I'll run the test under the race detector and see what it says.
